### PR TITLE
boostification of reqsession part1

### PR DESCRIPTION
**Pull request by sima-fastly**
manually tested

### CODE REVIEW

File: src/cpp/proxy/proxysession.cpp [hunk 1]
```diff
@@ -238,12 +249,14 @@ class ProxySession::Private : public QObject
...
-		connect(rs, &RequestSession::paused, this, &Private::rs_paused);
-		connect(rs, &RequestSession::headerBytesSent, this, &Private::rs_headerBytesSent);
-		connect(rs, &RequestSession::bodyBytesSent, this, &Private::rs_bodyBytesSent);
+		reqSessionConnectionMap[rs] = {
+			rs->bytesWritten.connect(boost::bind(&Private::rs_bytesWritten, this, boost::placeholders::_1, rs)),
+			rs->errorResponding.connect(boost::bind(&Private::rs_errorResponding, this, rs)),
+			rs->paused.connect(boost::bind(&Private::rs_paused, this, rs)),
+			rs->headerBytesSent.connect(boost::bind(&Private::rs_headerBytesSent, this, boost::placeholders::_1, rs)),
+			rs->bodyBytesSent.connect(boost::bind(&Private::rs_bodyBytesSent, this, boost::placeholders::_1, rs))
+		};
```
Comment (jkarneges) [inline src/cpp/proxy/proxysession.cpp]: are these cleaned up anywhere?

File: src/cpp/proxy/requestsession.h [hunk 2]
```diff
@@ -103,21 +104,20 @@ class RequestSession : public QObject
...
-	void paused();
-	void headerBytesSent(int count);
-	void bodyBytesSent(int count);
-
+	boost::signals2::signal<void(const InspectData&)> inspected;
+    Signal finishedByAccept;
+    SignalInt bytesWritten;
+    Signal paused;
+    SignalInt headerBytesSent;
+    SignalInt bodyBytesSent;
```
Comment (jkarneges) [inline src/cpp/proxy/requestsession.h:113]: indentation

File: src/cpp/proxy/proxysession.cpp (hunk 1 above)
Comment (sima-fastly) [reply src/cpp/proxy/proxysession.cpp:259]: true , updated to include logic

File: out.txt
[the remark this answers is not in the document]
Comment (sima-fastly) [reply out.txt:1]: yes indeed

**Review by jkarneges — APPROVED** [review]

**Pull request merged**